app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.java

```diff
@@ -163,10 +167,60 @@ public void showBooks(LinkedList<Book> books) {
     } else {
       libraryAdapter.getFilter().filter("", i -> stopScanningContent());
     }
-    libraryAdapter.notifyDataSetChanged();
+    onRefreshDisableListClick();
+    noRefreshQueryChange();
+    notifyDatachange();
     libraryList.setOnItemClickListener(this);
   }
 
+  public  void notifyDatachange(){
+    // we will notify the update every 0.3 second as the listitem position and content changes
+    final Handler handler = new Handler();
+    handler.postDelayed( new Runnable() {
+
+      @Override
+      public void run() {
+        activity.runOnUiThread(() -> libraryAdapter.notifyDataSetChanged());
+        handler.postDelayed( this, 300 );
+      }
+    }, 300 );
+  }
+  public void onRefreshDisableListClick(){
+    //  we will disable list while refreshing to prevent it from IndexOutofBoundException & IllegalException (notifyDataSetChanged())
+    if(swipeRefreshLayout.isRefreshing()){
```

Comment:
Are you sure the function defintion is in accordance with function name. I think what you intended is following:
`if(swipeRefreshLayout.isRefreshing()){
 libraryList.setEnabled(false);
}`

And there should not be else part otherwise on refreshing second time, it becomes enabled again.

Reply:
yes sure i will remove else part in my next commit